contracts/0.4.24/nos/NodeOperatorsRegistry.sol

```diff
@@ -973,53 +973,52 @@ contract NodeOperatorsRegistry is AragonApp, IStakingModule, Versioned {
         return TYPE_POSITION.getStorageBytes32();
     }
 
-    /// @notice Returns the validators stats of all node operators in the staking module
-    /// @return exitedValidatorsCount Total number of validators in the EXITED state
-    /// @return activeValidatorsKeysCount Total number of validators in active state
-    /// @return readyToDepositValidatorsKeysCount Total number of validators ready to be deposited
-    function getValidatorsKeysStats()
-        external
-        view
-        returns (
-            uint256 exitedValidatorsCount,
-            uint256 activeValidatorsKeysCount,
-            uint256 readyToDepositValidatorsKeysCount
-        )
-    {
+    // struct ValidatorsReport {
+    //     uint256 totalExited;
+    //     uint256 totalDeposited;
+    //     uint256 totalVetted;
+    //     uint256 totalStuck;
+    //     uint256 totalRefunded;
+    //     uint256 targetLimit;
+    //     uint256 excessCount;
+    // }
+
+    function getValidatorsReport() external view {
         SigningKeysStats.State memory totalSigningKeysStats = _getTotalSigningKeysStats();
-
-        uint256 vettedSigningKeysCount = totalSigningKeysStats.vettedSigningKeysCount;
-        uint256 depositedSigningKeysCount = totalSigningKeysStats.depositedSigningKeysCount;
-
-        exitedValidatorsCount = totalSigningKeysStats.exitedSigningKeysCount;
-        activeValidatorsKeysCount = depositedSigningKeysCount - exitedValidatorsCount;
-        readyToDepositValidatorsKeysCount = vettedSigningKeysCount - depositedSigningKeysCount;
+        bytes memory validatorsReportEncoded = abi.encode(
+            uint256(totalSigningKeysStats.exitedSigningKeysCount),
+            uint256(totalSigningKeysStats.depositedSigningKeysCount),
+            uint256(totalSigningKeysStats.vettedSigningKeysCount),
+            0, // totalStuckValidators
+            0, // totalRefundedValidators
+            0, // excessValidators
+            0 // targetLimit
+        );
+        assembly {
+            return(add(validatorsReportEncoded, 32), mul(32, 7))
```

Comment:
a comment explaining the magic is needed here

Reply:
Added comment with an explanation of what data is returned